cdc/redo/writer/file_allocator.go

```diff
@@ -0,0 +1,101 @@
+//  Copyright 2022 PingCAP, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package writer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pingcap/tiflow/cdc/redo/common"
+	"github.com/pingcap/tiflow/pkg/fsutil"
+)
+
+// ref: https://github.com/etcd-io/etcd/pull/4785
+// fileAllocator has two functionalities:
+// 1. create new file or reuse the existing file in advance
+//    before it will be written.
+// 2. pre-allocate disk space to mitigate the overhead of file metadata updating.
+type fileAllocator struct {
```

Comment:
Please consider making the struct public `FileAllocator` and put it in `pkg/falloc` or something similar? It can be useful for generic purposes.

Reply:
Good advise. It is moved to `pkg/fsutil` now.